src/components/AvatarWithDisplayName.tsx

```diff
@@ -1,114 +1,104 @@
         return;
     }
 
-    if (ReportUtils.isIOUReport(report)) {
+    if (ReportUtils.isIOUReport(report) && report?.reportID) {
         Navigation.navigate(ROUTES.REPORT_PARTICIPANTS.getRoute(report.reportID));
         return;
     }
 
     if (ReportUtils.isChatThread(report)) {
         const parentReportAction = ReportActionsUtils.getParentReportAction(report);
-        const actorAccountID = lodashGet(parentReportAction, 'actorAccountID', -1);
+        const actorAccountID = parentReportAction?.actorAccountID ?? -1;
         // in an ideal situation account ID won't be 0
         if (actorAccountID > 0) {
             Navigation.navigate(ROUTES.PROFILE.getRoute(actorAccountID));
             return;
         }
     }
 
-    // report detail route is added as fallback but based on the current implementation this route won't be executed
-    Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS.getRoute(report.reportID));
+    if (report?.reportID) {
+        // report detail route is added as fallback but based on the current implementation this route won't be executed
+        Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS.getRoute(report.reportID));
+    }
 };
 
-function AvatarWithDisplayName(props) {
+function AvatarWithDisplayName({
+    personalDetails,
+    policy,
+    report,
+    isAnonymous = false,
+    size = CONST.AVATAR_SIZE.DEFAULT,
+    shouldEnableDetailPageNavigation = false,
+}: AvatarWithDisplayNameProps) {
     const theme = useTheme();
     const styles = useThemeStyles();
-    const title = ReportUtils.getReportName(props.report);
-    const subtitle = ReportUtils.getChatRoomSubtitle(props.report);
-    const parentNavigationSubtitleData = ReportUtils.getParentNavigationSubtitle(props.report);
-    const isMoneyRequestOrReport = ReportUtils.isMoneyRequestReport(props.report) || ReportUtils.isMoneyRequest(props.report);
-    const icons = ReportUtils.getIcons(props.report, props.personalDetails, props.policy);
-    const ownerPersonalDetails = OptionsListUtils.getPersonalDetailsForAccountIDs([props.report.ownerAccountID], props.personalDetails);
-    const displayNamesWithTooltips = ReportUtils.getDisplayNamesWithTooltips(_.values(ownerPersonalDetails), false);
-    const shouldShowSubscriptAvatar = ReportUtils.shouldReportShowSubscript(props.report);
-    const isExpenseRequest = ReportUtils.isExpenseRequest(props.report);
-    const defaultSubscriptSize = isExpenseRequest ? CONST.AVATAR_SIZE.SMALL_NORMAL : props.size;
-    const avatarBorderColor = props.isAnonymous ? theme.highlightBG : theme.componentBG;
+    const title = ReportUtils.getReportName(report);
+    const subtitle = ReportUtils.getChatRoomSubtitle(report);
+    const parentNavigationSubtitleData = ReportUtils.getParentNavigationSubtitle(report);
+    const isMoneyRequestOrReport = ReportUtils.isMoneyRequestReport(report) || ReportUtils.isMoneyRequest(report);
+    const icons = ReportUtils.getIcons(report, personalDetails, null, '', -1, policy);
```

Comment:
I know this was implemented incorrectly, but make sure to test it, so that it doesn't make any regressions 😄